vtex-apps/mega-menu
# Feature/ssr load menu dom

**What problem is this solving?**

It loads the items of the mega menu both client side and server side when the page is initialized; this includes all departments, categories, subcategories. In other words, when the page loads, the mega menu items are already in the DOM. This is needed for SEO purposes.



A client requested this feature for SEO optimization.

**How should this be manually tested?**

The mega menu needs to be already set up, 
1. When the page loads, inspect the page and search for `vtex-mega-menu-2-x-styledLinkText` and you will see multiple matches. These are the menu items already loaded in the DOM.
2. Inspect the Page Source and do the same to verify that the same menu items are in the page source as well.

**Screenshots or example usage:**

1. Client side:
<img width="271" alt="image" src="https://user-images.githubusercontent.com/91942975/175930195-bdcbb98b-dd69-4641-b36b-f652193a9b99.png">
This mega menu has 1 department, 1 category and 1 subcategory. In the image you can see 1 of 3 so it loaded all the items.

2. Server side:
<img width="956" alt="image" src="https://user-images.githubusercontent.com/91942975/175930507-d993544d-34df-47ac-aad4-c6819859885e.png">
Same for the page source, 1 of 3.

This is the menu with 3 total items:
<img width="266" alt="image" src="https://user-images.githubusercontent.com/91942975/175930573-4b7eff6b-ad13-482c-8aef-57c6d7a29d65.png">
